src/util/enrichFunction.js

```diff
@@ -0,0 +1,17 @@
+import axios from "axios"
+
+async function enrichFunction(split, returnResp){
+    for(let i = 0 ;i<split.length;i++){
+        if(split[i] in returnResp){
+            const detailEntry = returnResp[split[i]]
+            for(let j = 0; j<detailEntry.length; j++){
+                const specificEntry = detailEntry[j]
+                const accessSpecificEntry = await axios.get(specificEntry)
```

Comment:
Por causa do for + await o seu algoritmo está fazendo todos os requests de maneira síncrona, ou seja, um após o outro.
Isso remove a característica mais importante de concorrência e velocidade do JS, que é capaz de fazer múltiplos downloads assíncronos de forma concorrente.

Dá uma lida em: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all